test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/DoSection.java

```diff
@@ -626,32 +628,55 @@ public String toString() {
         return result;
     }
 
+    private static boolean matchWithRange(String nodeVersionString, List<VersionRange> acceptedVersionRanges, XContentLocation location) {
+        var unqualifiedNodeVersionString = nodeVersionString.replace("-SNAPSHOT", "");
+        try {
+            Version version = Version.fromString(unqualifiedNodeVersionString);
+            return acceptedVersionRanges.stream().anyMatch(v -> v.contains(version));
+        } catch (IllegalArgumentException e) {
+            throw new XContentParseException(
+                location,
+                "[version] range node selector expects a semantic version format (x.y.z), but found " + unqualifiedNodeVersionString,
+                e
+            );
+        }
+    }
+
     private static NodeSelector parseVersionSelector(XContentParser parser) throws IOException {
         if (false == parser.currentToken().isValue()) {
             throw new XContentParseException(parser.getTokenLocation(), "expected [version] to be a value");
         }
-        List<VersionRange> skipVersionRanges = parser.text().equals("current")
-            ? List.of(new VersionRange(Version.CURRENT, Version.CURRENT))
-            : SkipSection.parseVersionRanges(parser.text());
+
+        final Predicate<String> nodeMatcher;
+        final String versionSelectorString;
+        if (parser.text().equals("current")) {
+            var currentUnqualified = Build.current().version().replace("-SNAPSHOT", "");
+            nodeMatcher = nodeVersion -> currentUnqualified.equals(nodeVersion.replace("-SNAPSHOT", ""));
+            versionSelectorString = "version is " + currentUnqualified + " (current)";
+        } else {
+            var acceptedVersionRange = SkipSection.parseVersionRanges(parser.text());
+            nodeMatcher = nodeVersion -> matchWithRange(nodeVersion, acceptedVersionRange, parser.getTokenLocation());
+            versionSelectorString = "version ranges " + acceptedVersionRange;
+        }
+
         return new NodeSelector() {
             @Override
             public void select(Iterable<Node> nodes) {
                 for (Iterator<Node> itr = nodes.iterator(); itr.hasNext();) {
                     Node node = itr.next();
-                    if (node.getVersion() == null) {
+                    String versionString = node.getVersion();
+                    if (versionString == null) {
                         throw new IllegalStateException("expected [version] metadata to be set but got " + node);
                     }
-                    Version version = Version.fromString(node.getVersion());
-                    boolean skip = skipVersionRanges.stream().anyMatch(v -> v.contains(version));
-                    if (false == skip) {
+                    if (nodeMatcher.test(versionString) == false) {
```

Comment:
If we do `replace("-SNAPSHOT", "")` here, the possible matchers both get simplified.

Reply:
Yes, it could become a simple string equals. Same as above, I will change it - it should work, and if not CI test failures will tell us.